### PR TITLE
Resolve config file references relative to config file, add initial config validation

### PR DESCRIPTION
**Pull request by liggitt**
@deads2k

### CODE REVIEW

File: pkg/cmd/server/api/helpers.go [hunk 1]
```diff
@@ -12,6 +13,82 @@ import (
...
+	refs = append(refs, &config.ServingInfo.ClientCA)
+
+	refs = append(refs, &config.MasterKubeConfig)
+
+	refs = append(refs, &config.VolumeDirectory)
+
+	return refs
+}
+
+func ResolvePaths(refs []*string, base string) error {
```
Comment (deads2k) [inline pkg/cmd/server/api/helpers.go]: This and relativize paths are both generic, so moving them to cmdutil makes sense.  We've already used similar logic in .kubeconfig file processing so we could collapse further into kube at some point.

File: pkg/cmd/server/api/helpers.go [hunk 2]
```diff
@@ -12,6 +13,82 @@ import (
 	"github.com/openshift/origin/pkg/cmd/server/crypto"
 )
 
+func GetMasterFileReferences(config *MasterConfig) []*string {
```
Comment (deads2k) [inline pkg/cmd/server/api/helpers.go]: Having a method called ResolveMasterConfigPaths that takes a masterconfig pointer will make it more obvious that mutation is happening.  It's not clear from calling code that these calls work together to mutate the underlying contents.

File: pkg/cmd/server/api/validation/validation.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	allErrs = append(allErrs, ValidateBindAddress(info.BindAddress)...)
+
+	if len(info.ServerCert.CertFile) > 0 {
+		if _, err := os.Stat(info.ServerCert.CertFile); os.IsNotExist(err) {
+			allErrs = append(allErrs, errs.NewFieldInvalid("certFile", info.ServerCert.CertFile, "does not exist"))
+		}
+
+		if len(info.ServerCert.KeyFile) == 0 {
+			allErrs = append(allErrs, errs.NewFieldRequired("keyFile", info.ServerCert.KeyFile))
+		} else if _, err := os.Stat(info.ServerCert.KeyFile); os.IsNotExist(err) {
```
Comment (deads2k) [inline pkg/cmd/server/api/validation/validation.go]: Any error should cause this to fail validation.

File: pkg/cmd/server/api/validation/validation.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+	if len(config.NodeName) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("nodeName", config.NodeName))
+	}
+
+	allErrs = append(allErrs, ValidateServingInfo(config.ServingInfo).Prefix("servingInfo")...)
+
+	if len(config.MasterKubeConfig) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("masterKubeConfig", config.MasterKubeConfig))
+	} else if _, err := os.Stat(config.MasterKubeConfig); os.IsNotExist(err) {
```
Comment (deads2k) [inline pkg/cmd/server/api/validation/validation.go]: again, any error reading should be an error.

**Comment by deads2k** [comment]
It's a good starting point.  Minor comments, once those are fixed its ready to merge.

**Comment by liggitt** [comment]
comments addressed, another commit added to put all the node certs (client and server) in a single dir in prep for node distribution

**Comment by deads2k** [comment]
You know, if you keep feeding it, it will get bigger....

**Comment by deads2k** [comment]
lgtm

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
@deads2k switched to validating only when we actually start the master/node (for now), otherwise you can't write out config without minting certs, and it makes unit testing harder

**Comment by liggitt** [comment]
[merge]

**Comment by yin** [comment]
[test]

**Pull request merged**